File: src/extras-attica/libknowledgebaseentry.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_Attica__KnowledgeBaseEntry
#include <QDateTime>
#include <QMap>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <knowledgebaseentry.h>
#include "libknowledgebaseentry.h"
#include "libknowledgebaseentry.hxx"

Attica__KnowledgeBaseEntry* Attica__KnowledgeBaseEntry_new() {
    return new Attica::KnowledgeBaseEntry();
}

Attica__KnowledgeBaseEntry* Attica__KnowledgeBaseEntry_new2(const Attica__KnowledgeBaseEntry* other) {
    return new Attica::KnowledgeBaseEntry(*other);
}

void Attica__KnowledgeBaseEntry_OperatorAssign(Attica__KnowledgeBaseEntry* self, const Attica__KnowledgeBaseEntry* other) {
    self->operator=(*other);
}

void Attica__KnowledgeBaseEntry_SetId(Attica__KnowledgeBaseEntry* self, libqt_string id) {
    QString id_QString = QString::fromUtf8(id.data, id.len);
    self->setId(id_QString);
}

libqt_string Attica__KnowledgeBaseEntry_Id(const Attica__KnowledgeBaseEntry* self) {
    QString _ret = self->id();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__KnowledgeBaseEntry_SetContentId(Attica__KnowledgeBaseEntry* self, int id) {
    self->setContentId(static_cast<int>(id));
}

int Attica__KnowledgeBaseEntry_ContentId(const Attica__KnowledgeBaseEntry* self) {
    return self->contentId();
}

void Attica__KnowledgeBaseEntry_SetUser(Attica__KnowledgeBaseEntry* self, const libqt_string user) {
    QString user_QString = QString::fromUtf8(user.data, user.len);
    self->setUser(user_QString);
}

libqt_string Attica__KnowledgeBaseEntry_User(const Attica__KnowledgeBaseEntry* self) {
    QString _ret = self->user();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__KnowledgeBaseEntry_SetStatus(Attica__KnowledgeBaseEntry* self, const libqt_string status) {
    QString status_QString = QString::fromUtf8(status.data, status.len);
    self->setStatus(status_QString);
}

libqt_string Attica__KnowledgeBaseEntry_Status(const Attica__KnowledgeBaseEntry* self) {
    QString _ret = self->status();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__KnowledgeBaseEntry_SetChanged(Attica__KnowledgeBaseEntry* self, const QDateTime* changed) {
    self->setChanged(*changed);
}

QDateTime* Attica__KnowledgeBaseEntry_Changed(const Attica__KnowledgeBaseEntry* self) {
    return new QDateTime(self->changed());
}

void Attica__KnowledgeBaseEntry_SetName(Attica__KnowledgeBaseEntry* self, const libqt_string name) {
    QString name_QString = QString::fromUtf8(name.data, name.len);
    self->setName(name_QString);
}

libqt_string Attica__KnowledgeBaseEntry_Name(const Attica__KnowledgeBaseEntry* self) {
    QString _ret = self->name();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__KnowledgeBaseEntry_SetDescription(Attica__KnowledgeBaseEntry* self, const libqt_string description) {
    QString description_QString = QString::fromUtf8(description.data, description.len);
    self->setDescription(description_QString);
}

libqt_string Attica__KnowledgeBaseEntry_Description(const Attica__KnowledgeBaseEntry* self) {
    QString _ret = self->description();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__KnowledgeBaseEntry_SetAnswer(Attica__KnowledgeBaseEntry* self, const libqt_string answer) {
    QString answer_QString = QString::fromUtf8(answer.data, answer.len);
    self->setAnswer(answer_QString);
}

libqt_string Attica__KnowledgeBaseEntry_Answer(const Attica__KnowledgeBaseEntry* self) {
    QString _ret = self->answer();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__KnowledgeBaseEntry_SetComments(Attica__KnowledgeBaseEntry* self, int comments) {
    self->setComments(static_cast<int>(comments));
}

int Attica__KnowledgeBaseEntry_Comments(const Attica__KnowledgeBaseEntry* self) {
    return self->comments();
}

void Attica__KnowledgeBaseEntry_SetDetailPage(Attica__KnowledgeBaseEntry* self, const QUrl* detailPage) {
    self->setDetailPage(*detailPage);
}

QUrl* Attica__KnowledgeBaseEntry_DetailPage(const Attica__KnowledgeBaseEntry* self) {
    return new QUrl(self->detailPage());
}

void Attica__KnowledgeBaseEntry_AddExtendedAttribute(Attica__KnowledgeBaseEntry* self, const libqt_string key, const libqt_string value) {
    QString key_QString = QString::fromUtf8(key.data, key.len);
    QString value_QString = QString::fromUtf8(value.data, value.len);
    self->addExtendedAttribute(key_QString, value_QString);
}

libqt_string Attica__KnowledgeBaseEntry_ExtendedAttribute(const Attica__KnowledgeBaseEntry* self, const libqt_string key) {
    QString key_QString = QString::fromUtf8(key.data, key.len);
    QString _ret = self->extendedAttribute(key_QString);
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_map /* of libqt_string to libqt_string */ Attica__KnowledgeBaseEntry_ExtendedAttributes(const Attica__KnowledgeBaseEntry* self) {
    QMap<QString, QString> _ret = self->extendedAttributes();
    // Convert QMap<> from C++ memory to manually-managed C memory
    libqt_string* _karr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    libqt_string* _varr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    int _ctr = 0;
    for (auto _itr = _ret.keyValueBegin(); _itr != _ret.keyValueEnd(); ++_itr) {
        QString _mapkey_ret = _itr->first;
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _mapkey_b = _mapkey_ret.toUtf8();
        libqt_string _mapkey_str;
        _mapkey_str.len = _mapkey_b.length();
        _mapkey_str.data = static_cast<const char*>(malloc(_mapkey_str.len + 1));
        memcpy((void*)_mapkey_str.data, _mapkey_b.data(), _mapkey_str.len);
        ((char*)_mapkey_str.data)[_mapkey_str.len] = '\0';
        _karr[_ctr] = _mapkey_str;
        QString _mapval_ret = _itr->second;
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _mapval_b = _mapval_ret.toUtf8();
        libqt_string _mapval_str;
        _mapval_str.len = _mapval_b.length();
        _mapval_str.data = static_cast<const char*>(malloc(_mapval_str.len + 1));
        memcpy((void*)_mapval_str.data, _mapval_b.data(), _mapval_str.len);
        ((char*)_mapval_str.data)[_mapval_str.len] = '\0';
        _varr[_ctr] = _mapval_str;
        _ctr++;
    }
    libqt_map _out;
    _out.len = _ret.size();
    _out.keys = static_cast<void*>(_karr);
    _out.values = static_cast<void*>(_varr);
    return _out;
}

bool Attica__KnowledgeBaseEntry_IsValid(const Attica__KnowledgeBaseEntry* self) {
    return self->isValid();
}

void Attica__KnowledgeBaseEntry_Delete(Attica__KnowledgeBaseEntry* self) {
    delete self;
}

File: include_original/extras-attica/libknowledgebaseentry.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");
pub const map_constu8_constu8 = std.StringHashMapUnmanaged([]const u8);

/// https://api.kde.org/attica-knowledgebaseentry.html
pub const attica__knowledgebaseentry = struct {
    /// New constructs a new Attica::KnowledgeBaseEntry object.
    ///
    ///
    pub fn New() QtC.Attica__KnowledgeBaseEntry {
        return qtc.Attica__KnowledgeBaseEntry_new();
    }

    /// New2 constructs a new Attica::KnowledgeBaseEntry object.
    ///
    /// ``` other: QtC.Attica__KnowledgeBaseEntry ```
    pub fn New2(other: ?*anyopaque) QtC.Attica__KnowledgeBaseEntry {
        return qtc.Attica__KnowledgeBaseEntry_new2(@ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#operator-eq)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, other: QtC.Attica__KnowledgeBaseEntry ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.Attica__KnowledgeBaseEntry_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setId)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, id: []const u8 ```
    pub fn SetId(self: ?*anyopaque, id: []const u8) void {
        const id_str = qtc.libqt_string{
            .len = id.len,
            .data = id.ptr,
        };
        qtc.Attica__KnowledgeBaseEntry_SetId(@ptrCast(self), id_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#id)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, allocator: std.mem.Allocator ```
    pub fn Id(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__KnowledgeBaseEntry_Id(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::knowledgebaseentry.Id: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setContentId)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, id: i32 ```
    pub fn SetContentId(self: ?*anyopaque, id: i32) void {
        qtc.Attica__KnowledgeBaseEntry_SetContentId(@ptrCast(self), @intCast(id));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#contentId)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry ```
    pub fn ContentId(self: ?*anyopaque) i32 {
        return qtc.Attica__KnowledgeBaseEntry_ContentId(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setUser)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, user: []const u8 ```
    pub fn SetUser(self: ?*anyopaque, user: []const u8) void {
        const user_str = qtc.libqt_string{
            .len = user.len,
            .data = user.ptr,
        };
        qtc.Attica__KnowledgeBaseEntry_SetUser(@ptrCast(self), user_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#user)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, allocator: std.mem.Allocator ```
    pub fn User(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__KnowledgeBaseEntry_User(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::knowledgebaseentry.User: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setStatus)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, status: []const u8 ```
    pub fn SetStatus(self: ?*anyopaque, status: []const u8) void {
        const status_str = qtc.libqt_string{
            .len = status.len,
            .data = status.ptr,
        };
        qtc.Attica__KnowledgeBaseEntry_SetStatus(@ptrCast(self), status_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#status)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, allocator: std.mem.Allocator ```
    pub fn Status(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__KnowledgeBaseEntry_Status(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::knowledgebaseentry.Status: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setChanged)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, changed: QtC.QDateTime ```
    pub fn SetChanged(self: ?*anyopaque, changed: ?*anyopaque) void {
        qtc.Attica__KnowledgeBaseEntry_SetChanged(@ptrCast(self), @ptrCast(changed));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#changed)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry ```
    pub fn Changed(self: ?*anyopaque) QtC.QDateTime {
        return qtc.Attica__KnowledgeBaseEntry_Changed(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setName)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, name: []const u8 ```
    pub fn SetName(self: ?*anyopaque, name: []const u8) void {
        const name_str = qtc.libqt_string{
            .len = name.len,
            .data = name.ptr,
        };
        qtc.Attica__KnowledgeBaseEntry_SetName(@ptrCast(self), name_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#name)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, allocator: std.mem.Allocator ```
    pub fn Name(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__KnowledgeBaseEntry_Name(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::knowledgebaseentry.Name: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setDescription)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, description: []const u8 ```
    pub fn SetDescription(self: ?*anyopaque, description: []const u8) void {
        const description_str = qtc.libqt_string{
            .len = description.len,
            .data = description.ptr,
        };
        qtc.Attica__KnowledgeBaseEntry_SetDescription(@ptrCast(self), description_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#description)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, allocator: std.mem.Allocator ```
    pub fn Description(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__KnowledgeBaseEntry_Description(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::knowledgebaseentry.Description: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setAnswer)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, answer: []const u8 ```
    pub fn SetAnswer(self: ?*anyopaque, answer: []const u8) void {
        const answer_str = qtc.libqt_string{
            .len = answer.len,
            .data = answer.ptr,
        };
        qtc.Attica__KnowledgeBaseEntry_SetAnswer(@ptrCast(self), answer_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#answer)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, allocator: std.mem.Allocator ```
    pub fn Answer(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__KnowledgeBaseEntry_Answer(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::knowledgebaseentry.Answer: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setComments)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, comments: i32 ```
    pub fn SetComments(self: ?*anyopaque, comments: i32) void {
        qtc.Attica__KnowledgeBaseEntry_SetComments(@ptrCast(self), @intCast(comments));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#comments)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry ```
    pub fn Comments(self: ?*anyopaque) i32 {
        return qtc.Attica__KnowledgeBaseEntry_Comments(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#setDetailPage)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, detailPage: QtC.QUrl ```
    pub fn SetDetailPage(self: ?*anyopaque, detailPage: ?*anyopaque) void {
        qtc.Attica__KnowledgeBaseEntry_SetDetailPage(@ptrCast(self), @ptrCast(detailPage));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#detailPage)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry ```
    pub fn DetailPage(self: ?*anyopaque) QtC.QUrl {
        return qtc.Attica__KnowledgeBaseEntry_DetailPage(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#addExtendedAttribute)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, key: []const u8, value: []const u8 ```
    pub fn AddExtendedAttribute(self: ?*anyopaque, key: []const u8, value: []const u8) void {
        const key_str = qtc.libqt_string{
            .len = key.len,
            .data = key.ptr,
        };
        const value_str = qtc.libqt_string{
            .len = value.len,
            .data = value.ptr,
        };
        qtc.Attica__KnowledgeBaseEntry_AddExtendedAttribute(@ptrCast(self), key_str, value_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#extendedAttribute)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, key: []const u8, allocator: std.mem.Allocator ```
    pub fn ExtendedAttribute(self: ?*anyopaque, key: []const u8, allocator: std.mem.Allocator) []const u8 {
        const key_str = qtc.libqt_string{
            .len = key.len,
            .data = key.ptr,
        };
        const _str = qtc.Attica__KnowledgeBaseEntry_ExtendedAttribute(@ptrCast(self), key_str);
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::knowledgebaseentry.ExtendedAttribute: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#extendedAttributes)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry, allocator: std.mem.Allocator ```
    pub fn ExtendedAttributes(self: ?*anyopaque, allocator: std.mem.Allocator) map_constu8_constu8 {
        const _map: qtc.libqt_map = qtc.Attica__KnowledgeBaseEntry_ExtendedAttributes(@ptrCast(self));
        var _ret: map_constu8_constu8 = .empty;
        defer {
            const _keys: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.keys));
            for (0.._map.len) |i| {
                qtc.libqt_free(_keys[i].data);
            }
            qtc.libqt_free(_map.keys);
            qtc.libqt_free(_map.values);
        }
        const _keys: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.keys));
        const _values: [*][]const u8 = @ptrCast(@alignCast(_map.values));
        var i: usize = 0;
        while (i < _map.len) : (i += 1) {
            const _key = _keys[i];
            const _entry_slice = std.mem.span(_key.data);
            const _value = _values[i];
            _ret.put(allocator, _entry_slice, _value) catch @panic("attica::knowledgebaseentry.ExtendedAttributes: Memory allocation failed");
        }
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-knowledgebaseentry.html#isValid)
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.Attica__KnowledgeBaseEntry_IsValid(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.Attica__KnowledgeBaseEntry ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.Attica__KnowledgeBaseEntry_Delete(@ptrCast(self));
    }
};
